dataservice/api/sample/resources.py

```diff
@@ -0,0 +1,120 @@
+from flask import (
+    abort,
+    request
+)
+from flask.views import MethodView
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound
+from marshmallow import ValidationError
+
+from dataservice.extensions import db
+from dataservice.api.errors import handle_integrity_error
+from dataservice.api.sample.models import Sample
+from dataservice.api.sample.schemas import SampleSchema
+
+
+class SampleAPI(MethodView):
+    """
+    Sample REST API
+    """
+
+    def get(self, kf_id):
+        """
+        Get a sample by id or get all samples
+
+        Get a sample by Kids First id or get all samples if
+        Kids First id is None
+        """
+        # Get all
+        if kf_id is None:
+            s = Sample.query.all()
+            return SampleSchema(many=True).jsonify(s)
+        # Get one
+        else:
+            try:
+                s = Sample.query.filter_by(kf_id=kf_id).one()
+            # Not found in database
+            except NoResultFound:
+                abort(404, 'could not find {} `{}`'
+                      .format('sample', kf_id))
+            return SampleSchema().jsonify(s)
+
+    def post(self):
+        """
+        Create a new sample
+        """
+        body = request.json
+        try:
```

Comment:
The schema validation and session transaction should be two different try/catches.

Reply:
Why? It would essentially be the same thing. Each line in the try except throws a very specific exception